compiler/p/codegen/OMRMemoryReference.cpp

```diff
@@ -1068,375 +1068,272 @@ void OMR::Power::MemoryReference::mapOpCode(TR::Instruction *currentInstruction)
 
-      if (self()->getUnresolvedSnippet() != NULL)
+      if (displacement == PTOC_FULL_INDEX)
          {
-#ifdef J9_PROJECT_SPECIFIC
-         getUnresolvedSnippet()->setAddressOfDataReference(cursor);
-         getUnresolvedSnippet()->setMemoryReference(self());
-         getUnresolvedSnippet()->setDataRegister(target);
-         cg->addRelocation(new (cg->trHeapMemory()) TR::LabelRelative24BitRelocation(cursor, getUnresolvedSnippet()->getSnippetLabel()));
-         *wcursor = 0x48000000;                        // b SnippetLabel;
-         wcursor++;
-         cursor += PPC_INSTRUCTION_LENGTH;
-
-         if (displacement != PTOC_FULL_INDEX)
-            {
-            if (displacement<LOWER_IMMED || displacement>UPPER_IMMED)
-               {
-               *wcursor = preserve | lower;
-               toRealRegister(getModBase())->setRegisterFieldRA(wcursor);      // mb is the modified base
-               cursor += PPC_INSTRUCTION_LENGTH;
-               }
-            }
-         else
-            {
-            *wcursor = 0x60000000;                    // ori target, target, bits16-31
-            target->setRegisterFieldRS(wcursor);
-            target->setRegisterFieldRA(wcursor);
-            wcursor++;
-            cursor += PPC_INSTRUCTION_LENGTH;
-
-            *wcursor = 0x780007c6;                    // rldicr target, target, 32, 31
-            target->setRegisterFieldRS(wcursor);
-            target->setRegisterFieldRA(wcursor);
-            wcursor++;
-            cursor += PPC_INSTRUCTION_LENGTH;
-
-            *wcursor = 0x64000000;                    // oris target, target, bits32-47
-            target->setRegisterFieldRS(wcursor);
-            target->setRegisterFieldRA(wcursor);
-            wcursor++;
-            cursor += PPC_INSTRUCTION_LENGTH;
-
-            *wcursor = 0x60000000;                    // ori target, target, bits48-63
-            target->setRegisterFieldRS(wcursor);
-            target->setRegisterFieldRA(wcursor);
-            wcursor++;
-            cursor += PPC_INSTRUCTION_LENGTH;
-            }
-#endif
+         prevInstruction = generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::ori, node, data, data, 0, prevInstruction);
+         prevInstruction = generateTrg1Src1Imm2Instruction(cg, TR::InstOpCode::rldicr, node, data, data, 32, 0xffffffff00000000ULL, prevInstruction);
+         prevInstruction = generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::oris, node, data, data, 0, prevInstruction);
+         prevInstruction = generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::ori, node, data, data, 0, prevInstruction);
+         }
+      else if (displacement < LOWER_IMMED || displacement > UPPER_IMMED)
+         {
+         newMR = new (cg->trHeapMemory()) TR::MemoryReference(self()->getModBase(), LO_VALUE(displacement), self()->getLength(), cg);
+         }
+      }
+   else if (index != NULL || isUsingDelayedIndexedForm())
+      {
+      if (isUsingDelayedIndexedForm())
```

Comment:
I think we used to check for index !=NULL first and then isUsingDelayedindexedForm. If these conditions are not mutually exclusive that can make a difference.

Reply:
Fixed in ed1deb4b7baaa9b3ec3af440fc146d44f7446d2b